src/praisonai-agents/praisonaiagents/agent/tool_execution.py

```
@@ -1307,6 +1307,28 @@ async def _async_approval():
+        the tool is not denied, preserving backward compatibility.
+        """
+        manager = getattr(self, "_permission_manager", None)
+        if manager is None:
+            return None
+        try:
+            if manager.is_denied(function_name, getattr(self, "name", None)):
+                return {
+                    "error": f"Tool '{function_name}' blocked by permission policy",
+                    "permission_denied": True,
+                }
+        except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)